### PR TITLE
add optional render argument to SubprocVecEnv

### PR DESCRIPTION
**Pull request by ethanabrooks**
Adds an optional `render` argument to `SubprocVecEnv` and `DummyVecEnv` which renders each step, including terminal steps.

## Description

- adds a `render` argument to aforesaid classes
- adds a `render_mode` argument to aforesaid classes
- renders each step (including terminal) if `render == True`

## Motivation and Context

In order to render the terminal step of an episode, `.render()` must be called within the `.step` method of `VecEnv`. For some environments it is important to be able to view the terminal step for debugging purposes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist:

- [x] I've read the [CONTRIBUTION](https://github.com/DLR-RM/stable-baselines3/blob/master/CONTRIBUTING.md) guide (**required**)
- [x] I have updated the changelog accordingly (**required**).
- [x] My change requires a change to the documentation.
- [x] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format` (**required**)
- [x] I have checked the codestyle using `make check-codestyle` and `make lint` (**required**)
- [x] I have ensured `make pytest` 
- [ ]  and `make type` both pass. (**required**) -- this threw an error before my changes at `stable_baselines3.common.base_class`
- [x] I have checked that the documentation builds using `make doc` (**required**)

Note: You can run most of the checks using `make commit-checks`.

Note: we are using a maximum length of 127 characters per line

### CODE REVIEW

File: stable_baselines3/common/vec_env/subproc_vec_env.py [hunk 1]
```diff
@@ -120,7 +141,12 @@ def step_wait(self) -> VecEnvStepReturn:
         results = [remote.recv() for remote in self.remotes]
         self.waiting = False
         obs, rews, dones, infos = zip(*results)
-        return _flatten_obs(obs, self.observation_space), np.stack(rews), np.stack(dones), infos
+        return (
```
Comment (ethanabrooks) [inline stable_baselines3/common/vec_env/subproc_vec_env.py:144]: This change came from `make format`. No logic added.

File: stable_baselines3/common/vec_env/subproc_vec_env.py [hunk 2]
```diff
@@ -168,7 +194,13 @@ def set_attr(self, attr_name: str, value: Any, indices: VecEnvIndices = None) ->
...
             remote.recv()
 
-    def env_method(self, method_name: str, *method_args, indices: VecEnvIndices = None, **method_kwargs) -> List[Any]:
+    def env_method(
+        self,
+        method_name: str,
+        *method_args,
+        indices: VecEnvIndices = None,
+        **method_kwargs,
+    ) -> List[Any]:
```
Comment (ethanabrooks) [inline stable_baselines3/common/vec_env/subproc_vec_env.py:203]: This change came from `make format`. No logic added.

File: stable_baselines3/common/vec_env/subproc_vec_env.py [hunk 3]
```diff
@@ -120,7 +141,12 @@ def step_wait(self) -> VecEnvStepReturn:
         results = [remote.recv() for remote in self.remotes]
         self.waiting = False
         obs, rews, dones, infos = zip(*results)
-        return _flatten_obs(obs, self.observation_space), np.stack(rews), np.stack(dones), infos
+        return (
+            _flatten_obs(obs, self.observation_space),
+            np.stack(rews),
+            np.stack(dones),
+            infos,
+        )
```
Comment (ethanabrooks) [inline stable_baselines3/common/vec_env/subproc_vec_env.py:149]: This change came from `make format`. No logic added.

File: stable_baselines3/common/vec_env/dummy_vec_env.py [hunk 4]
```diff
@@ -40,15 +52,24 @@ def step_async(self, actions: np.ndarray) -> None:
...
                 self.buf_infos[env_idx]["terminal_observation"] = obs
                 obs = self.envs[env_idx].reset()
             self._save_obs(env_idx, obs)
-        return (self._obs_from_buf(), np.copy(self.buf_rews), np.copy(self.buf_dones), deepcopy(self.buf_infos))
+        return (
+            self._obs_from_buf(),
+            np.copy(self.buf_rews),
+            np.copy(self.buf_dones),
+            deepcopy(self.buf_infos),
+        )
```
Comment (ethanabrooks) [inline stable_baselines3/common/vec_env/dummy_vec_env.py:72]: This change came from `make format`. No logic added.

**Comment by araffin** [comment]
Hello,

as mentioned in the contributing guide and in the PR template, please open an issue first with the appropriate issue template.

>For some environments it is important to be able to view the terminal step for debugging purposes.

this sounds like something that can be solved within the custom environment, by passing a keyword argument at creation time or setting a flag to true using `set_attr()` method.

**Comment by araffin** [comment]
I will close this as an alternative solution has been proposed in https://github.com/DLR-RM/stable-baselines3/issues/282#issuecomment-754896849

**Pull request closed without merging**